lib/rubocop/cop/rspec/missing_expectation_target_method.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks if the ExpectationTarget methods `.to`, `not_to` or `to_not` are
+      # used.
+      #
+      # @example
+      #   # bad
+      #   expect(something).kind_of? Foo
+      #   is_expected == 42
+      #
+      #   # good
+      #   expect(something).to be_a Foo
+      #   is_expected.to eq 42
+      #
+      class MissingExpectationTargetMethod < Base
+        MSG = 'Use `.to`, `.not_to` or `.to_not` to set an expectation.'
+        RESTRICT_ON_SEND = %i[expect is_expected].freeze
+
+        # @!method expect?(node)
+        def_node_matcher :expect?, <<~PATTERN
+          {
+            (send nil? :expect ...)
+            (send nil? :is_expected)
+          }
+        PATTERN
+
+        # @!method expect_block?(node)
+        def_node_matcher :expect_block?, <<~PATTERN
+          (block #expect? (args) _body)
+        PATTERN
+
+        # @!method expectation_without_runner?(node)
+        def_node_matcher :expectation_without_runner?, <<~PATTERN
+          (send {#expect? #expect_block?} !#Runners.all ...)
+        PATTERN
+
+        def on_send(node)
+          if node.parent&.block_type?
```

Comment:
```suggestion
         add_offense(node.loc.selector) if expectation_without_runner?(node)
       end
```
And remove everything below.
Would it work like that?

Reply:
Is that because we want to highlight the offence differently for the block vs value expectations?

Reply:
I don’t know. My suggested code works like the code suggested in the PR (the specs pass without change).

Reply:
Sorry @pirj, I only just now understood what you meant.

Your simpler solution is indeed possible, but I think we’d have to remove `RESTRICT_ON_SEND`.

Reply:
value expectations:

```
$ ruby-parse -e 'expect(foo).to' 
(send
  (send nil :expect
    (send nil :foo)) :to)
```

with block:

```
$ ruby-parse -e 'expect{foo}.to' 
(send
  (block
    (send nil :expect)
    (args)
    (send nil :foo)) :to)
```

In both cases I wanted to highlight the `to` location. Is there a better way?
The specs did not pass with `expectation_without_runner?(node)`.

Reply:
@bquorning Now when you mentioned RESTRICT_ON_SEND, i started understanding why it is implemented the way it is.
No objections either way.
I thought that a concise nodepattern-only implementation is superior, but with restrict on send the choice is not that trivial.
I’m wondering if we allow our users to configure expectations in the dsl, and if we should account for that. That’s my only argument left for the nodepattern solution.